frontend/app/lib/api.ts

```diff
@@ -1,28 +1,37 @@
-import { apiUrl } from "../utils/env";
+import { apiUrl } from "../utils/env"
 
 async function fetchWithAuth(endpoint: string, options: RequestInit = {}) {
-  const headers = new Headers(options.headers);
+  if (!apiUrl) {
+    throw new Error("API URL is not configured")
+  }
```

Comment:
Since we are providing a default value when the api is not provided, this shouldn't be necessary, but it doesn't harm.